### PR TITLE
Cumulative develop branch update

### PR DESCRIPTION
**Pull request by sorpaas**

### CODE REVIEW

**Review by athei — APPROVED** [review]
LGTM but I would really appreciate if features would stay additive.

File: src/lib.rs [hunk 1]
```diff
@@ -36,6 +37,18 @@ use crate::{
     util::{Decoder, SignatureArray},
 };
 
+#[cfg(all(feature = "static-context", feature = "lazy-static-context"))]
+compile_error!("Should only enable one of static-context or lazy-static-context");
```
[the remark this answers is not in the document]
Comment (sorpaas) [reply src/lib.rs:41]: Those two feature flags are never meant to be used together. If that happens, the combination is meaningless and it's a config error of the packages using it.

[the remark this answers is not in the document]
Comment (athei) [reply src/lib.rs:41]: The thing is that this "meaningless" combination isn't the fault of any packages using this crate. Imagine this dependency graph ( X -> Y means X depends on Y):

```
A -> B
A -> C
B -> libsecp256k1(static_context)
C -> libsecp256k1(lazy_static_context)
```

The way that feature unification works in cargo is that a single version of `libsecp256k1` is included in the dependency tree with both features enabled. It isn't the fault of `A` that its dependency use incompatible features.

[the remark this answers is not in the document]
Comment (sorpaas) [reply src/lib.rs:41]: I see. We need to choose which overwrites which though. I'll take a look (mostly whether lazy_static works good enough on no_std) to resolve this.

**Comment by sorpaas** [comment]
Please don't merge this yet. We need to enable merge commit for this repo and merge it that way.

**Pull request merged**